python/cudf/cudf/api/types.py

```diff
@@ -27,6 +27,35 @@
 )
 
 
+def dtype(arbitrary):
+    try:
+        np_dtype = np.dtype(arbitrary)
+        if np_dtype.name == "float16":
+            np_dtype = np.dtype("float32")
+        elif np_dtype.name in ("object", "str"):
+            np_dtype = np.dtype("object")
+        elif np_dtype.str == "<m8":
+            np_dtype = np.dtype("<m8[ns]")
+        elif np_dtype.str == "<M8":
+            np_dtype = np.dtype("<M8[ns]")
+        return np_dtype
+    except TypeError:
+        pass
+    if isinstance(arbitrary, cudf.core.dtypes._BaseDtype):
+        return arbitrary
+    elif isinstance(arbitrary, pd.CategoricalDtype):
+        return cudf.CategoricalDtype.from_pandas(arbitrary)
+    elif isinstance(arbitrary, pd.IntervalDtype):
+        return cudf.IntervalDtype.from_pandas(arbitrary)
+    pd_dtype = pd.api.types.pandas_dtype(arbitrary)
+    try:
+        return pd_dtype.numpy_dtype
+    except AttributeError:
+        # no NumPy type corresponding to this type
+        # always object?
+        return np.dtype("object")
```

Comment:
This seems reasonable to me, as pandas is moving towards `object` as default type if no dtype is provided:
```python
>>> pd.Series()
<stdin>:1: DeprecationWarning: The default dtype for empty Series will be 'object' instead of 'float64' in a future version. Specify a dtype explicitly to silence this warning.
Series([], dtype: float64)
```